scrapy_zyte_api/handler.py

```diff
@@ -86,6 +114,149 @@ def _prepare_api_params(self, request: Request) -> Optional[dict]:
                 f"instead. ({request})."
             )
             raise
+
+        if request.meta.get("zyte_api_automap", self._automap):
+            self._map_params(api_params, request)
+
+        return api_params
+
+    def _map_headers(self, api_params, request):
```

Comment:
Could you try converting _map_headers and _map_params to some utility methods, which work without a need to create a handler? It might simplify testing.